### PR TITLE
Parse XML errors like in S3

### PR DESCRIPTION
**Pull request by cenedhryn**
Changes the definition of the document error root to what is used in AWS S3, so that errors are fully parsed. 

## Motivation and Context
To this point, errors returned from S3Control have lost their XML content and only the error type has been returned, showing null: 
~~~
software.amazon.awssdk.services.s3control.model.S3ControlException: null
~~~
With this change, errors are parsed correctly as:
~~~
software.amazon.awssdk.services.s3control.model.S3ControlException: Missing role arn 
(Service: S3Control, Status Code: 400, Request ID: 4T5H9GEV7H3K9MCJ, Extended Request ID: USkN9APZoEVT0...)
~~~

## Testing
- Unit tests and functional tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/s3control/src/main/resources/log4j.properties [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#
```
Comment (dagnir) [inline services/s3control/src/main/resources/log4j.properties:1]: We shouldn't add `log4j.properties` in the main resources

File: services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java [hunk 2]
```diff
@@ -81,7 +86,7 @@ private String resolveHost(SdkHttpRequest request, String accountId, S3ControlCo
 
         }
         validateComponentIsHostnameCompliant(accountId, "account id");
-        return String.format("%s.%s", accountId, host);
+        return hasEndpointOverride ? host : String.format("%s.%s", accountId, host);
```
Comment (dagnir) [inline services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java:89]: Thinking about this some more, I'm not sure that this is the right behavior; it's possible that the user could still want this behavior even with an endpoint override.

Can we leave this change out of the client for now? For testing, we can we can just add a new client interceptor that changes the request URL to the WireMock host before it's sent.

Comment (cenedhryn) [reply services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java:89]: Yes, that sounds like a less intrusive change.

File: services/s3control/src/main/resources/log4j.properties (hunk 1 above)
Comment (cenedhryn) [reply services/s3control/src/main/resources/log4j.properties:1]: I thought I added it in the test directory, will move it.

File: services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java (hunk 2 above)
Comment (dagnir) [inline services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java:89]: New changes look good, but looks like this still needs to be reverted along with the accompanying test.

**Review by dagnir — APPROVED** [review]
:shipit:

**Pull request merged**